### ISSUE TITLE
[RE-OPENED] In Global Fonts, use markup defaults for font options.

### ISSUE THREAD

**Issue by gravnetic** [elementor/elementor#12535]
**_Primary, Secondary, Text, Base...these are very confusing and terrible UX.  If the text is 'text' then what is Primary, it also text?  Is Secondary 'Text' or is it some other font that is not text?_**

1.  Please consider updating the Global Type options to use semantic definitions.  The WC3 exists for a reason.  The random markup selectors you have chosen only compound confusion.

2.  I would like you to be able to disable and HIDE these unusual options for the type if non-sematic definitions are going to remain.

3.  Please provide the option to have both REM and PX active, so I can set a base font size on html that will be set by px and allow for REM overwrites.  This is beneficial for browser font scaling.

![screenshot](https://user-images.githubusercontent.com/475331/93220831-40702580-f72a-11ea-8b1f-cb3bd4385616.jpg)

**Comment by dasigna** [comment]
second this another time.

the choosen definitions are _really_ confusing - especially if one defines custom globals according to markup or alike and they are still there to choose from for editors e.g. ...
they cant be set to any logic for not confusing them.

the given definitions might make sense if theres a style guide available that uses similar expressions by chance.
not to mention those definitions are staying in css anywys ... should they?

**Comment by black-eye** [comment]
+1

**Comment by dlayd** [comment]
Tl;dr: the names relate to the widget elements. 

I've actually had to ask this question before and got an answer from elementor. They relate to the widgets themselves, I believe primary is for the widget titles, secondary is for body, accent is something else, etc. 

I think it serves a purpose, it just needs to have a better definition/documentation.

**Comment by dasigna** [comment]
> 
> 
> Tl;dr: the names relate to the widget elements.
> 
> I've actually had to ask this question before and got an answer from elementor. They relate to the widgets themselves, I believe primary is for the widget titles, secondary is for body, accent is something else, etc.
> 
> I think it serves a purpose, it just needs to have a better definition/documentation.

would have been easy to communicate it somewhere (or at least explain it here).

although does not make much more sense IMO. maybe just use it as an additional (internal) 'widgets-option' that is not necessarily reflected in css ...?
besides that - i am not aware that this is picked up for any default widget settings at all. seems to me they remain as before - so wheres the benefit?

**Comment by dlayd** [comment]
> > Tl;dr: the names relate to the widget elements.
> > I've actually had to ask this question before and got an answer from elementor. They relate to the widgets themselves, I believe primary is for the widget titles, secondary is for body, accent is something else, etc.
> > I think it serves a purpose, it just needs to have a better definition/documentation.
> 
> would have been easy to communicate it somewhere (or at least explain it here).
> 
> although does not make much more sense IMO. maybe just use it as an additional (internal) 'widgets-option' that is not necessarily reflected in css ...?
> besides that - i am not aware that this is picked up for any default widget settings at all. seems to me they remain as before - so wheres the benefit?

I think it's an artifact from the initial build. Global colors/fonts weren't standardized for web dev, it was more of a custom solution for them which allowed users to apply pseudo-global styles to the elements rather than have to individually style each one. 

That's just my impression. I've been using Elementor since mid-2018. It's time for it to go, or get a re-vamp.

**Comment by samheavyside** [comment]
I have to say, I was super excited by the idea of Global Fonts and Colours, but at this stage, I am just terribly disappointed and confused by the whole Global "Fonts" thing.

The problem, as I see it, is that currently there is an "All or nothing" approach to the application of Global Fonts. For example, if you want to change the size of a particular element with a Global Font assigned, the Global Font link will be broken and any changes to the GF definition in the future will not be applied to the element. That's just wrong.

Surely the Font-Family is the only thing that _needs_ to be defined in a Global Font. Sure, it could possibly be handy to define a custom fallback font-stack, and if you're loading Google Fonts, you should be able to choose which font weights are loaded, but beyond that, the Global "Font" definition should NOT contain any other stylisation properties (and in particular the size). 

Those stylistic choices (weight, size, line-height, letter-spacing, color etc) should be handled on the element itself, or separately in another section called "Global Styles" or something. 

Let's say I have 3 Font Families that I want to use on my site. One for "Headings", one for "Body" text and a "Fancy" one for limited uses. 

I want to be able to easily apply these fonts to any/every element and then proceed to apply any specific styling to the element (size, colour, font-weight, italics) that particular instance demands - or - choose to apply a predefined "Global Style" which sets all those things. 

If I decide that the main "Heading" font I have chosen for the site needs changing, then I should be able to make a new selection in _1 place_ (ie. at the Global Font definition) and have that font change be reflected throughout the entire site, irrespective of what specific styles have been applied to each element. 

Furthermore, if I have been using a predefined "Global Style" to bring uniformity to various elements, I should be able to easily update the style in 1 place and have those changes reflected across the entire site.

The idea that I need to individually define the same font-family I'm using for every different combination of font size, weight etc etc that appears on my site, it just seems really poorly thought out. 

Thanks for listening to my rant.

**Comment by gravnetic** [comment]
### Please Consider ASAP, Big Opportunity to Do The Right Thing Here - What would Spike Lee do?

**@shilo-ey I really hope that rather than just close this ticket you will work with a developer and consider what I am saying.  Elementor has an obligation to ditch widget names that make no sense and translate them to sematic tags.**

**@dlayd - thank you for that insight.  I have to stand my ground that once Elementor jumped into globals they have a responsibility to use semantic names and values.  I would even argue that they could hint towards best practice enabling preferred options as defaults.  For example it I were to turn on Global fonts why not set a default base font stack, a global font in px, and your defined font tags as rem units.  Then you are actually teaching designers best practices while adding a ton of value.**

**Comment by samheavyside** [comment]
One other question I have regarding "Global Fonts" - or just "Font" management in general. 

Can someone explain about how the "Disable Default Fonts" checkbox setting works (in Dashboard > Elementor > Settings > General)?

The explanation of what this setting (if enabled) does is:
_"Checking this box will disable Elementor's Default Fonts, and make Elementor inherit the fonts from your theme."_

I am using Hello Theme. If it is left Unchecked, I would assume this means I am using Elementor's Default Fonts. I would also assume that the default fonts are specified in the Site Settings / Global Fonts area, right? Well, when trying to setup those "Default Fonts" I am presented with the following notice in the Site Settings > Typography panel:

_"In order for **Theme Style** to affect all relevant Elementor elements, please disable Default Colors and Fonts from the Settings Page."_

Isn't that the opposite of what one would expect? The wording alongside the checkbox is would suggest that by Checking that option, the _Theme_ (in this case, Hello Theme, but could easily be GeneratePress, or OceanWP etc etc) would then take over control of the Font Settings.  But instead it seems that you need to Check the box in order to have access to the Elementor "Theme Style" settings? What? 

Side note: I think the terminology "Theme Style" being used in the Site Settings panel is extremely confusing - especially if you are using a theme which does have it's own styling settings (usually through the customizer).

Layer that on top of the other "Global Font vs. Typography" rubbish mentioned previously, I am left wondering "why is this SO darn confusing and hard??"

**Comment by dasigna** [comment]
... we're leaving the initial thread a bit - but yeah, every thing depend from another - does it? :)

so the "disable default fonts" settings ...

if you _dont_ check them -regard less of theme- elementor picks up the default stiles of the editor widgets. (fonts and colors) which are whatever elementor sets it to be.

if you check these boxes, the styles set within your **wp** theme are taking place. e.g. from generatepres, ocean wp, etc. with 'hello theme' you still wont have no theme style.
as this was before the global styling options of elementor, the term 'theme' for these checkboxes obviously doesnt refer to that. (although elementor names its new global settings 'theme style' which leads to another confusion)

now - if you have checked and dont set anything within **EL** theme style, basically the **WP** theme styles should take place (still default elementor if using a blank theme like 'hello').
at this point theres another confusion: as the **wp** theme font settings seem to be left untouched by elementor theme this isnt the case for colors - the do appear in css nevertheless. where used? probably for widgets (hence the names) but i dont see them taking place when altered ... opposite for **EL** theme style settings for fonts - they obviously do take place "globally".

so the 'elementor theme styles' mimics the 'wp theme' and will overwrite any settings of this (if set).
at this point for us all to see not that well. programmed confusion :)

it seems you have to check those mentioned boxes even if using 'hello' to let elementor know it might use its 'global settings' aka 'theme style'? dunno.
confusing? yes.


so yes, one has to fiddle its way through this, although it basically isnt that complicated and hasnt to be at all.
most confusion in fact is generated by the namings of elementor itself... so e.g. 'theme style' within elementor editor settings refer to 'elementor _global_ styles' (which 'mimics' the theme in a way and overrides the 'wp theme'). 'theme style' in description for checkboxes refers to the 'wp theme' ...
this is sometimes even more confusing when having weird or incorrect translations from someone who isnt even a native speaker and uses google or whatever to translate.


maybe that might help to sort things out? or maybe i am completely confused too and facing misundertstadings?

anyway - this has to be sorted out and cleaned. thoroughly.

as i stated before somewhere recently: elementor IMO has to pause that marketing things, and clean the house up before inviting new customers again.
or at least publish blog posts where they explain such mundane things... :)

**Comment by samheavyside** [comment]
@dasigna, I agree this has departed from the original thread topic, so I will start a new thread more specific to my complaints :)

Apologies to @gravnetic for inadvertently hijacking the thread. Your concern over the semantic naming of font "styles" is still an issue and needs addressing too (as well as the somewhat separate issue of REM unit fallbacks).
